library-best-practices/glossary.md

```diff
@@ -66,3 +66,77 @@ Java Dependency Glossary
+
+
+### Static Linkage Check
+
+- **Static linkage check**: a process to identify the existence of static
+  linkage errors for a given classpath, by scanning Java classes and
+  verifying the availability in the classpath.
+
+- **Static linkage report**: the outcome of a static linkage check. It reports
+  zero or more static linkage errors.
+
+- **Static linkage errors**: a reported error for a reference as the result of
+  a static linkage check.
+  A static linkage error contains the information on the source class and
+  the destination class of the reference. Each static linkage error has
+  one of the three types:
+
+  - _Missing class type_ is the error when where the destination class of a
+    class reference does not exist in the linkage classpath. This error
+    happens when a class is removed in a different version of a library,
+    or there is a dependency missed when constructing the linkage classpath.
+    The reference that causes a missing class error is called a _dangling reference_.
+
+  - _Missing method type_ is the error when a method reference has a  static
+    linkage conflict.
+
+  - _Missing field type_ is the error when a field reference has a static
+     linkage conflict.
+
+- **Linkage classpath**: the classpath which a static linkage check
+  runs against. For a Maven project, a linkage classpath consists of the
+  project's class files and the transitive dependencies induced by the direct
+  dependencies in the project's pom.xml file.
+
+- **Class usage graph**: a directed graph that represents the relationship between
+  classes in the linkage classpath. The nodes (vertex) of the graph correspond to
+  Java classes and the directed edges are references between classes.
+  For example, when 'Class A' has reference to 'Class B' on calling a method X,
+  the class usage graph holds an edge between two nodes:
+
+  ```
+  [Class A] --(method X of class B)-> [Class B]
+  ```
+
+  In this case, 'Class A' is called the _source class_ of the reference and
+  'Class B' is called the _destination class_.
+
+- **A reference**: in the class usage graph, the relationship between two 
+  classes is either _class reference_, _method reference_ or _field reference_.
+
+  A _method reference_ indicates that the source class uses the (static or
+  non-static) method of the destination class.
+
+  A _field reference_ indicates that the source class accesses the (static or
+  non-static) field of the destination class.
+
+  A _class reference_ indicates that the source class uses the destination
+  class without referencing a specific field or method (e.g., class inheritance).
+
+  A reference is called to have a _linkage conflict_
+  when there is a static linkage conflict caused by the reference.
+
+- **Reachability** is the attribute of static linkage errors to indicate
+  whether a linkage error caused by a reference is _reachable_ from the _entry
+  point classes_ of the class usage graph. In other words, when a static
```

Comment:
entry point classes are separate from the class usage graph, not part of it.

Reply:
Removed from "of the class usage graph".